compliance_checker/suite.py

```diff
@@ -695,9 +695,22 @@ def generate_dataset(self, cdl_path):
             ds_str = cdl_path.replace('.cdl', '.nc')
         else:
             ds_str = cdl_path + '.nc'
-        returncode = subprocess.call(['ncgen', '-o', ds_str, cdl_path])
-        if returncode != 0:
-            print("Error generating dataset from .cdl")
+
+        # generate netCDF-4 file
+        iostat = subprocess.run(['ncgen', '-k', 'nc4', '-o', ds_str, cdl_path], capture_output=True)
+        if iostat.returncode != 0:
+          # if not successfull, create netCDF classic file
+          print('netCDF-4 file could not be generated from cdl file with ' +
+                'message:')
+          print(iostat.stderr)
+          print('Trying to create netCDF Classic file instead.')
+          iostat = subprocess.run(['ncgen', '-k', 'nc3', '-o', ds_str, cdl_path], capture_output=True)
+          if iostat.returncode != 0:
+            # Exit program if neither a netCDF Classic nor a netCDF-4 file
+            # could be created.
+            print('netCDF Classic file could not be generated from cdl file' +
+                  'with message:')
+            print(iostat.stderr)
```

Comment:
The `capture_output` keyword is only available in Python 3.7+, but this Compliance Checker needs to support Python 3.5+. If you supply `subprocess.PIPE` as the `stderr` keywrod, you'll be able to keep almost everything the same but maintain compatibility:

```
iostat = subprocess.run(['ncgen', '-k', 'nc4', '-o', ds_str, cdl_path], stderr=subprocess.PIPE)
...
print(iostat.stderr.decode()) # stored as byte string
...
```

If the output is not captured, an exception is thrown upon failure. At first I wanted to use this functionality and wrap in try-except statements, but I think this is a better idea. Pinging @benjwadams to see if he has any input here.

Reply:
Thanks for the info. I updated the code accordingly.